### PR TITLE
Core/Spells: Fix Seed of Corruption (Warlock) target selection

### LINKED ISSUE #26010: Seed of corruption's explosion damages through walls and structures.
**Description:**

Seed of corruption's explosion will hit creatures in other rooms through walls, this is especially bad in dungeons, a swarm of creatures aggroed from another room will run at you.

Here's an example in scholomance it hits the skeletons in a room under the ground
image

**Expected behaviour:**

Seed of corruption should not hit mobs behind walls or out of LOS

**Steps to reproduce the problem:**

1. spawn a mob anywhere with another mob behind a wall, or udner it, make seed explode, see if it damages the other mobs. Or just try it in scholomance

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

92c8ffc047ea35ea9a534a878cd4be5726f20974

**Operating system:** 

windows

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

-  Patch 2.3.0 (2007-11-13): Seed of Corruption detonation will now obey line of sight: https://wow.gamepedia.com/Seed_of_Corruption
-  Seed of Corruption detonation should also hit seed of corruption actionTarget (this was working fine because GetExplTargetUnit() can't get that target here, I only remove that unnecessary check): http://www.worldoflogs.com/reports/ak6ponc2cou5kufz/xe/?s=1062&e=1349&x=spell+%3D+%22Seed+of+Corruption%22%0D%0AAND+sourceName+%3D+%22Plazma%22

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [ ] master

**Issues addressed:**

Closes #26010

**Tests performed:**

tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_warlock.cpp [hunk 1]
```diff
@@ -1147,31 +1147,27 @@ class spell_warl_seduction : public SpellScriptLoader
...
         {
-            PrepareSpellScript(spell_warl_seed_of_corruption_SpellScript);
-
-            void FilterTargets(std::list<WorldObject*>& targets)
-            {
-                if (GetExplTargetUnit())
-                    targets.remove(GetExplTargetUnit());
-            }
+            if (Unit const* unitTarget = target->ToUnit())
+                if (WorldLocation* dest = const_cast<WorldLocation*>(GetExplTargetDest()))
```
Comment (jackpoz) [inline src/server/scripts/Spells/spell_warlock.cpp:1159]: do you need the const_cast for something in particular here ?

**Comment by jackpoz** [comment]
does the LoS exclusion actually need a spell script ? isn't there an out of the box way to achieve the same ? or would that be considered a "hack" etc ?

**Comment by Jildor** [comment]
can remove:
```
AttributesEx2: 0x00000004 (SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS)
```
 but it would surely be hacky, so I have decided to filter by spellscript, I don't know if there would be a more proper way

File: src/server/scripts/Spells/spell_warlock.cpp (hunk 1 above)
Comment (Jildor) [reply src/server/scripts/Spells/spell_warlock.cpp:1159]: copypaste from: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp#L2294

**Comment by offl** [comment]
```cpp
    // Seed of Corruption
    ApplySpellFix({ 27285 }, [](SpellInfo* spellInfo)
    {
        spellInfo->AttributesEx2 &= ~SPELL_ATTR2_CAN_TARGET_NOT_IN_LOS;
    });
```

**Comment by Jildor** [comment]
@offl As I have indicated to jackpoz, I think that would be considered hacky, if you remove that attribute, what happens if the caster is out of LoS when the proc occurs?  Maybe the aoe would not be launched in the actiontarget position as it is not in LoS with caster?

**Comment by offl** [comment]
I'm not really a spell stuff guy
Just pointed to possible scenario in case you meant something different by 'can remove'

File: src/server/scripts/Spells/spell_warlock.cpp (hunk 1 above)
Comment (mdX7) [reply src/server/scripts/Spells/spell_warlock.cpp:1159]: in the linked case the WorldLocation gets relocated, we're just getting the coords here => const_cast not required

Comment (Jildor) [reply src/server/scripts/Spells/spell_warlock.cpp]: yep sorry, I couldn't make the change yesterday, thanks to both

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**